movie/src/main/java/com/calvinnor/movie/commons/ui/HomeActivity.kt

```diff
@@ -1,6 +1,28 @@
 package com.calvinnor.movie.commons.ui
 
+import android.os.Bundle
+import android.view.View.GONE
+import android.view.View.VISIBLE
+import androidx.core.view.isGone
+import androidx.core.view.isVisible
+import androidx.lifecycle.Observer
 import com.calvinnor.core.ui.BaseActivity
 import com.calvinnor.movie.R
+import com.calvinnor.movie.commons.util.NetworkUtils.getNetworkLiveData
+import kotlinx.android.synthetic.main.activity_home.*
 
-class HomeActivity : BaseActivity(R.layout.activity_home)
+class HomeActivity : BaseActivity(R.layout.activity_home) {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        getNetworkLiveData(applicationContext).observe(this, Observer { isConnected ->
```

Comment:
Nit: We have an extension for LiveData that creates the observer for you. You can simply do

```
getNetworkLiveData(applicationContext).observe(this) {
    // code block
}
```